apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -159,30 +129,37 @@ export class CredentialsService implements CredentialVerifier {
     if (authenticateDto.options.proofPurpose !== ProofPurpose.authentication) {
       throw new BadRequestException('proof purpose must be authentication for DIDAuth');
     }
+    const signPresentationDto: ProvePresentationDto = {
+      presentation: {
+        '@context': ['https://www.w3.org/2018/credentials/v1'],
+        type: ['VerifiablePresentation', 'DidAuth'],
+        holder: authenticateDto.did
+      },
+      options: authenticateDto.options
+    };
 
-    const verificationMethodId =
-      authenticateDto.options.verificationMethod ??
-      (await this.getVerificationMethodForDid(authenticateDto.did)).id;
-
-    const key = await this.getKeyForVerificationMethod(verificationMethodId);
-    const proofOptions = this.mapVcApiPresentationOptionsToSpruceIssueOptions(authenticateDto.options);
+    const signed = await this.provePresentation(signPresentationDto);
 
-    return didKitExecutor<VerifiablePresentationDto>(
-      () => DIDAuth(authenticateDto.did, JSON.stringify(proofOptions), JSON.stringify(key)),
-      'DIDAuth'
-    );
+    return signed;
   }
 
   async verifyPresentation(
     vp: VerifiablePresentationDto,
     options: VerifyOptionsDto
   ): Promise<VerificationResultDto> {
-    const verifyOptions: ISpruceVerifyOptions = options;
-
-    return didKitExecutor<VerificationResultDto>(
-      () => verifyPresentation(JSON.stringify(vp), JSON.stringify(verifyOptions)),
-      'verifyPresentation'
+    const w3cVerifyPresentationOptions: W3cVerifyPresentationOptions = {
+      presentation: JsonTransformer.fromJSON(vp, W3cJsonLdVerifiablePresentation, {
+        // Credo is expecting the the verifiableCredential property in a VP
+        // However, this property is optional in both the v1.1 and v2 VC data models
+        // For example, it isn't present if the VP is only for authentication
+        validate: !(vp.verifiableCredential === undefined)
+      }),
```

Comment:
Better to use resuable function IMO. See https://github.com/impactility/vc-api/pull/5